### DISCUSSION TITLE
Need help to figure out the sudden stop dequeue of control queue

### DISCUSSION THREAD

**Discussion by siemenstan** [Azure/azure-functions-durable-extension#2357] [Q&A]
Hi,
I would like to get some help how to figure out the reason of sudden stop dequeue of control queue at 12-Jan-2023 19:03:47 UTC. My app is using durable function extension v2.5.1, and deployed into Azure Kubernetes. From the App Insights, I am able to see the trace of "Stopped listening for messages on queue" but there not seeing any exception or trace message that indicate the reason of the sudden stop, appreciate if anyone can suggest what is any other traces that can help to figure out root cause of such issue. Thank you in advance.

Below are some error info and screenshot of app insights trace:
Azure Region: West US
Timestamp (UTC): 2023-01-12T19:03:47.4552886Z
HostInstanceId: a4ffcd8d-4b6e-4192-b645-bbbdb3a1ddf7
appId: 8f2c3085-e255-4aa8-bb45-175531af3e9e
Function App: C# .NET 6.0 with Durable function extension v2.5.1

![image](https://user-images.githubusercontent.com/98590735/212244149-83c2c585-bd2b-42d6-8326-fbfa39c03eb6.png)

**Comment by cgillum** [comment]
There are usually two reasons for an app instance to stop listening for messages on a control queue:
1. The functions host is shutting down.
2. One of the partitions has been stolen by / rebalanced to another instance.

If you're observing that you have app instances running and there are no app instances reading from a particular queue, then it's possible you've encountered a bug that can cause app instances to unexpectedly stop listening for queue messages. I recommend you update to the latest extension version (currently v2.9.0) to make sure you have all the bug fixes we've made in this area to rule out this possibility.

**Reply by siemenstan** [reply]
Thank you @cgillum 

The function host is still running and the partition is doesn't seems to be served by any other function host (the queue count in that particular partition keep increasing), eventually we restarted the kubernetes deployment to restart the app host. Could you please share any trace or pattern that help me to find out
1. Reason (or any hint) that can indicate why the stop of the queue listening.
2. Check whether the partition queue is being served or assigned to any instance?

We are going to upgrade to the extension v2.6.1, but upgrade to any newer version is currently being hold due to the [issue ](https://github.com/Azure/azure-functions-durable-extension/issues/2336) reported earlier

**Reply by davidmrdavid** [reply]
Checking whether the queue is being served by another instance should be easier. See if you can find a log such as "Started listening for messages on queue <your queue name / PartitionId here>" in another worker/instance shortly after you see the "Stopped listening for messages ..." log. That should tell you whether (and when) another worker took over that queue.

As for determining why we stopped listening to a queue, that's a bit harder. At the risk of doing too broad of a search: if you do a wide query for the string "shutdown" in your traces, do you see the worker/instance that stopped listening to the queue receiving a shutdown request around the time it stopped listening for messages?

**Reply by siemenstan** [reply]
Checked that, there is no "Started listening for messages on queue" nor "shutdown" message around 19:03:47 UTC. "Starting listening for messages" trace seen only after the production team restarted the app service around 20:47 UTC since we pick up high queue count and delays

Below is the app insights screenshot
![image](https://user-images.githubusercontent.com/98590735/213607706-e8d8c7d2-97d9-4e1e-93ca-c7b6b47c6c6a.png)

**Reply by davidmrdavid** [reply]
Interesting. It does appear _to me_ like no other worker picked up the control queue until the restart.

I'm also surprised that we see no logs for "shutdown" since a restart would count as one. I wonder if the search is case sensitive, as I noticed you capitalized "shutdown" in your query. I'll defer again to @cgillum for further advice.

**Reply by siemenstan** [reply]
I did query with "`shutdown`" first, but surprised to see there is no _shutdown_ trace nor _stopped listening for messages_ upon the app containers restart, that why I run the query with "`Shutdown`" after that.

**Reply by davidmrdavid** [reply]
I see. So just so I'm clear: you found no logs with `"shutdown"` but you found results with `"Shutdown"`, correct?
Also a ping to @cgillum in case he has anything to add here.
@siemenstan: have there been any developments / new findings since your last update or not?

**Reply by siemenstan** [reply]
@davidmrdavid  No, I didn't find any shutdown trace with "`shutdown`" or "`Shutdown`" keyword. There is no findings on the reason of the sudden dequeue stop, and the app has been running ok after the restart so far.
